packages/hash/api/src/graphql/resolvers/ontology/data-type.ts

```diff
@@ -0,0 +1,57 @@
+import { ApolloError } from "apollo-server-express";
+import { AxiosError } from "axios";
+
+import {
+  IdentifiedDataType,
+  QueryGetDataTypeArgs,
+  Resolver,
+} from "../../apiTypes.gen";
+import { GraphQLContext } from "../../context";
+import { DataTypeModel } from "../../../model";
+import { NIL_UUID } from "../../../model/util";
+
+export const getAllLatestDataTypes: Resolver<
+  Promise<IdentifiedDataType[]>,
+  {},
+  GraphQLContext,
+  {}
+> = async (_, __, { dataSources }) => {
+  const { graphApi } = dataSources;
+
+  const allLatestDataTypeModels = await DataTypeModel.getAllLatest(graphApi, {
+    /** @todo Replace with User from the request */
+    accountId: NIL_UUID,
```

Comment:
This would prevent a user from accessing types in another account using this method. Should this mutation therefore be called something like `getAllLatestDataTypesInMyAccount`?

Reply:
I see this more as getting all latest data types that the user has access to view, and the `accountId` from the request would be used for authorization

Reply:
Ah okay, if that's how we're interpreting `accountId` for this method than this makes sense to me - thanks

Reply:
(I think we're generally going to need to revisit this because we interpret `accountId` differently depending on the places)

As we find a way to more publicly expose the query builder approach, I think we'll be more clear about authentication v using identities as a filter.

Reply:
We're probably going to encounter some issues from this in the meantime, but I think that's just a compromise we'll have to deal with